### PR TITLE
Grant kibana_system role read access to APM data streams

### PR DESCRIPTION
**Pull request by dgieselaar**
The kibana_system role currently has access to classic APM indices (apm-*) for data telemetry purposes. When APM Server is running under Fleet, as is now the default, APM data is written to data streams that match a different pattern. While we updated the privileges of the apm_user role, we neglected to do so for the kibana_system role, leading to a gap in our telemetry.

This change adds the same privileges to the kibana_system role as we've previously granted to the apm_user role.

See:
elastic/apm-dev#731
https://github.com/elastic/elasticsearch/pull/68749
https://github.com/elastic/elasticsearch/pull/52917

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -704,6 +704,12 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                     .build(),
                 // APM telemetry queries APM indices in kibana task runner
                 RoleDescriptor.IndicesPrivileges.builder().indices("apm-*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("logs-apm.*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("logs-apm-*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("metrics-apm.*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("metrics-apm-*").privileges("read", "read_cross_cluster").build(),
```
Comment (Mpdreamz) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:710]: Given our datastreams [docs](https://www.elastic.co/guide/en/apm/guide/8.1/apm-data-streams.html) and [code](https://github.com/elastic/apm-server/tree/main/apmpackage/apm/data_stream)
```
Application traces: traces-apm-<namespace>
RUM and iOS agent application traces: traces-apm.rum-<namespace>
APM internal metrics: metrics-apm.internal-<namespace>
APM profiling metrics: metrics-apm.profiling-<namespace>
Application metrics: metrics-apm.app.<service.name>-<namespace>
logs-apm.error-<namespace>
```

```diff
apm-*
logs-apm.*
logs-apm-*"
metrics-apm.*
-metrics-apm-*
traces-apm.*
traces-apm-*
```

I don't think we strictly need access to the `apm` dataset name under the `metrics` type (`{type}-{dataset}-{namespace}`)

**Review by Mpdreamz — CHANGES_REQUESTED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java (hunk 1 above)
Comment (dgieselaar) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:710]: @Mpdreamz does the same apply to `logs-apm-*`? I don't see that in your list either.

**Review by azasypkin — APPROVED** [review]
Based on the outcome of the discussions related to `kibana_system` having a `read` access to some data indices solely for the telemetry purposes - LGTM from the Kibana Security side (assuming `logs-apm-*` is removed from the PR, if not needed).

* [Data Telemetry for Elastic](https://github.com/elastic/dev/issues/1405)
* [[APM] Allow kibana to collect APM telemetry in background task (conclusion)](https://github.com/elastic/elasticsearch/pull/52917#discussion_r393765136)

**Review by Mpdreamz — APPROVED** [review]
LGTM :+1:

**Review by ywangd — APPROVED** [review]
LGTM

**Comment by dgieselaar** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.2|[](https://github.com/elastic/elasticsearch/pull/85923)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**